### PR TITLE
Update branch protection settings.

### PR DESCRIPTION
**Pull request by geeknoid**
- Add support for a number of our repos.

- Standardize on a common release-* for our release branches.

- Disable tide stuff on istio/api to transition to mergify.

### CODE REVIEW

File: prow/plugins.yaml
Comment (fejta) [inline prow/plugins.yaml]: Best practice here is to trigger at the `istio` and `istio-releases` org level, rather than each repo individually.

Historically that would have caused issues if the repos didn't have any prow jobs (I think comment spam). It is no longer the case. 

This is how we configure the kubernetes org:
* https://github.com/kubernetes/test-infra/blob/d4a5312e1597a3be2d8f0ae96ca4c40e94b092a0/prow/plugins.yaml#L5-L12
* https://groups.google.com/d/msg/kubernetes-dev/D8_6DtuhyS4/IQHYNXqRBwAJ

File: prow/plugins.yaml
Comment (fejta) [inline prow/plugins.yaml]: Do we want to include the label sync change in this PR?

File: prow/config.yaml
Comment (fejta) [inline prow/config.yaml:54]: a) [Tide blockers](https://github.com/kubernetes/test-infra/tree/master/prow/tide/blockers) is a better way to achieve this effect. Ask @cjwagner about it. The [blockade plugin](https://github.com/kubernetes/test-infra/tree/master/prow/plugins/blockade) is something else to consider.

b) I suspect admins simply press the merge button? The [override plugin](https://github.com/kubernetes/test-infra/blob/master/prow/plugins/override/override.go) is probably a better alternative to use here if you're not already aware of it.

c) yaml anchors tend to make it harder to understand these config files, which limits the number of people willing to maintain them. Best to avoid them and inline the values each time. Use a script to generate the file when/if this becomes onerous.

File: prow/config.yaml
Comment (fejta) [inline prow/config.yaml:78]: What branches do we want to leave unprotected and why?

File: prow/plugins.yaml
Comment (fejta) [inline prow/plugins.yaml]: What is the intent here?

File: prow/plugins.yaml
Comment (geeknoid) [reply prow/plugins.yaml]: Thanks, updated.

File: prow/config.yaml
Comment (geeknoid) [reply prow/config.yaml:78]: The non-official branches are left unprotected. So branches developers create in an ad-hoc fashion as part of doing a PR. We discourage the practice, but it's common nevertheless.

File: prow/plugins.yaml
Comment (geeknoid) [reply prow/plugins.yaml]: Like most of our other repos, istio/api and istio/istio are being moved to the mergify model for handling merges and stuff, and we want people to just use the GitHub UI in the normal way.

File: prow/plugins.yaml
Comment (geeknoid) [reply prow/plugins.yaml]: Whoops, nope.

File: prow/config.yaml
Comment (geeknoid) [reply prow/config.yaml:54]: As mentioned previously, most of our repos don't use tide and we're moving istio/api and istio/istio off of tide as well. So I don't believe the plugins you suggest  would work for us in that context.

As for the yaml anchors, I think it substantially increases readability and maintainability, so I'd like to keep 'em as-is. For most of our repos so far, very few humans have been dealing with branch protection stuff (i.e. me), so I'm not worried about a constraint on people not being familiar with the syntax.

**Comment by geeknoid** [comment]
PTAL

**Review by fejta — COMMENTED** [review]
Can you pull the plugins.yaml changes into their own PR? They need further discussion.

Other things LGTM

**Comment by geeknoid** [comment]
Done, changes to plugins.yaml were removed.

File: prow/config.yaml
Comment (fejta) [inline prow/config.yaml:210]: Can we also move the tide changes to their own PR?

**Comment by geeknoid** [comment]
Given our discussion, can I check this in as-is?

Thanks.

**Review by fejta — APPROVED** [review]
Sure SGTM, merge at your leisure

/hold

**Comment by fejta** [comment]
Prefer a `/hold cancel` over manual merge

**Comment by fpesce** [comment]
Yeah, what just happened here @geeknoid ?

**Pull request merged**
